hudi-flink/src/test/java/org/apache/hudi/operator/MergeOnReadCompactTest.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.operator;
+
+import org.apache.flink.configuration.Configuration;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Test cases for delta stream write with compaction.
+ */
+public class MergeOnReadCompactTest extends CopyOnWriteTest {
```

Comment:
for units tests and test cases, seems that should be independent of each other.
```
MergeOnReadWriteTest extends CopyOnWriteTest
MergeOnReadCompactTest extends CopyOnWriteTest
```
for base content, shall we use a base test? like
```
AbstractFlinkWriteTest
CopyOnWriteTest extends AbstractFlinkWriteTest
MergeOnReadWriteTest extends AbstractFlinkWriteTest
MergeOnReadCompactTest extends AbstractFlinkWriteTest
```

Reply:
Sorry, i think there is no need to have another base class here, because the base class should have some the full test cases for `COPY_ON_WRITE` table type.

Reply:
CopyOnWriteTest、MergeOnReadWriteTest、MergeOnReadCompactTest share same tests, can use `@ParameterizedTest` to test (`MERGE_ON_READ`, `COPY_ON_WRITE`) ?

Reply:
It is not very convenient because we have some custom strategies for each type, that means the code are not 100% reused, IMHO, `ParameterizedTest` is more suitable for values enumeration.